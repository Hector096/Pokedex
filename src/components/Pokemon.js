import {
  Card, Image, Tag, List,
} from 'antd';
import getColor from '../color';

const Pokemon = ({ data }) => {
  const { Meta } = Card;
  return (
    <div>
      <Card
        cover={(
          <div className="d-flex justify-content-center">
            <Image
              width={200}
              src={`${data.sprites.front_default}`}
            />
          </div>
    )}
      >
        <Meta
          title={data.name.toUpperCase()}
          className="mb-2"
        />
        <span>
          {data.types.map((type) => {
            const color = getColor(type.type.name);
            return (
              <Tag color={color} key={type.type.url}>
                {type.type.name.toUpperCase()}
              </Tag>
            );
          })}
        </span>
        <List
          dataSource={data.stats}
          renderItem={(item) => (
            <List.Item key={item.stat.name}>
              <List.Item.Meta
                title={item.stat.name}
              />
              <div>{item.base_stat}</div>
            </List.Item>
          )}
        />
      </Card>
    </div>
  );
};
export default Pokemon;
