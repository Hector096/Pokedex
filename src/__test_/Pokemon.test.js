import TestRenderer from 'react-test-renderer';
import Pokemon from '../components/Pokemon';

const data = {
  base_experience: 142,
  height: 11,
  id: 5,
  name: 'charmeleon',
  sprites: { front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/5.png' },
  stats: [
    {
      base_stat: 58,
      effort: 0,
      stat: {
        name: 'hp',
        url: 'https://pokeapi.co/api/v2/stat/1/',
      },
    },
    {
      base_stat: 64,
      effort: 0,
      stat: {
        name: 'attack',
        url: 'https://pokeapi.co/api/v2/stat/2/',
      },
    },
    {
      base_stat: 58,
      effort: 0,
      stat: {
        name: 'defense',
        url: 'https://pokeapi.co/api/v2/stat/3/',
      },
    },
    {
      base_stat: 80,
      effort: 1,
      stat: {
        name: 'special-attack',
        url: 'https://pokeapi.co/api/v2/stat/4/',
      },
    },
    {
      base_stat: 65,
      effort: 0,
      stat: {
        name: 'special-defense',
        url: 'https://pokeapi.co/api/v2/stat/5/',
      },
    },
    {
      base_stat: 80,
      effort: 1,
      stat: {
        name: 'speed',
        url: 'https://pokeapi.co/api/v2/stat/6/',
      },
    },
  ],
  types: [
    {
      slot: 1,
      type: {
        name: 'fire',
        url: 'https://pokeapi.co/api/v2/type/10/',
      },
    },
  ],
  weight: 190,
};
describe('component test', () => {
  test('matches Pokemon snapshot', () => {
    const tree = TestRenderer.create(
      <Pokemon data={data} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
