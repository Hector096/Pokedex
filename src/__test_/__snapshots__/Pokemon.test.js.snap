// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component test matches Pokemon snapshot 1`] = `
<div>
  <div
    className="ant-card ant-card-bordered"
  >
    <div
      className="ant-card-cover"
    >
      <div
        className="d-flex justify-content-center"
      >
        <div
          className="ant-image"
          onClick={[Function]}
          style={
            Object {
              "height": undefined,
              "width": 200,
            }
          }
        >
          <img
            className="ant-image-img"
            onError={[Function]}
            onLoad={[Function]}
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/5.png"
            style={
              Object {
                "height": undefined,
              }
            }
          />
          <div
            className="ant-image-mask"
          >
            <div
              className="ant-image-mask-info"
            >
              <span
                aria-label="eye"
                className="anticon anticon-eye"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="eye"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                  />
                </svg>
              </span>
              Preview
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="ant-card-meta mb-2"
      >
        <div
          className="ant-card-meta-detail"
        >
          <div
            className="ant-card-meta-title"
          >
            CHARMELEON
          </div>
        </div>
      </div>
      <span>
        <span
          className="ant-tag ant-tag-has-color"
          style={
            Object {
              "backgroundColor": "#F08030",
            }
          }
        >
          FIRE
        </span>
      </span>
      <div
        className="ant-list ant-list-split"
      >
        <div
          className="ant-spin-nested-loading"
        >
          <div
            className="ant-spin-container"
          >
            <ul
              className="ant-list-items"
            >
              <li
                className="ant-list-item"
              >
                <div
                  className="ant-list-item-meta"
                >
                  <div
                    className="ant-list-item-meta-content"
                  >
                    <h4
                      className="ant-list-item-meta-title"
                    >
                      hp
                    </h4>
                  </div>
                </div>
                <div>
                  58
                </div>
              </li>
              <li
                className="ant-list-item"
              >
                <div
                  className="ant-list-item-meta"
                >
                  <div
                    className="ant-list-item-meta-content"
                  >
                    <h4
                      className="ant-list-item-meta-title"
                    >
                      attack
                    </h4>
                  </div>
                </div>
                <div>
                  64
                </div>
              </li>
              <li
                className="ant-list-item"
              >
                <div
                  className="ant-list-item-meta"
                >
                  <div
                    className="ant-list-item-meta-content"
                  >
                    <h4
                      className="ant-list-item-meta-title"
                    >
                      defense
                    </h4>
                  </div>
                </div>
                <div>
                  58
                </div>
              </li>
              <li
                className="ant-list-item"
              >
                <div
                  className="ant-list-item-meta"
                >
                  <div
                    className="ant-list-item-meta-content"
                  >
                    <h4
                      className="ant-list-item-meta-title"
                    >
                      special-attack
                    </h4>
                  </div>
                </div>
                <div>
                  80
                </div>
              </li>
              <li
                className="ant-list-item"
              >
                <div
                  className="ant-list-item-meta"
                >
                  <div
                    className="ant-list-item-meta-content"
                  >
                    <h4
                      className="ant-list-item-meta-title"
                    >
                      special-defense
                    </h4>
                  </div>
                </div>
                <div>
                  65
                </div>
              </li>
              <li
                className="ant-list-item"
              >
                <div
                  className="ant-list-item-meta"
                >
                  <div
                    className="ant-list-item-meta-content"
                  >
                    <h4
                      className="ant-list-item-meta-title"
                    >
                      speed
                    </h4>
                  </div>
                </div>
                <div>
                  80
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
